tests/ui/impl-trait/issue-55872.rs

```diff
@@ -10,8 +10,8 @@ impl<S> Bar for S {
     type E = impl Copy;
 
     fn foo<T>() -> Self::E {
-        || ()
         //~^ ERROR type parameter `T` is part of concrete type but not used in parameter list for the `impl Trait` type alias
+        || ()
```

Comment:
unfortunate side-effect from HIR typeck eagerly replacing the return type with an inference var. This will be partially fixed when removing this eager normalization call. The new solver does point to the closure instead.

However, as we're still normalizing away opaques to infer vars during generalization, there may still be span regressions where we point at the place introducing the opaque instead of the place which actually constrains it. This feels unavoidable without major changes to the way we track spans for opaques/in the type system in general.

Reply:
OK, yeah that seems like the major fallout of this PR (along w/ the bad spans for mismatched hidden types).

I guess most of these will not be a problem w/ RPIT.